boostercloud/booster
# feat(cli): make resource templates publishable

## Description
This PR is adding new `boost stub:publish` command that will improve DX by allowing developers to publish and customise resource template files 🗄
![image](https://user-images.githubusercontent.com/36774784/168446817-7f90f4e7-ebc5-47f2-836c-bd74cb4ff62e.png)
> Do you see that "look content from updated stub! 🐙" ? 👀  
It will now appear in every new command file!

### Usage
```bash 
$ boost stub:publish 
```
Will create `/stubs` folder in project root and copy each resource [template](https://github.com/boostercloud/booster/tree/main/packages/cli/src/templates) file. If you have published resources before, you will have to confirm overwriting the files, or you can pass the `--force` flag to bypass this.

#### QA
<details>
  <summary>Can I have only one stub for a certain resource?</summary>
  
  Yes! The resource generator will check if you have a custom template or it will use the default template
</details>

<details>
  <summary>How can I keep up with new template updates?</summary>
  
  1. Run `boost stub:publish --force` command
  2. Review changes
  3. done!
</details>

<details>
  <summary>Can I adjust the command template and leave the other resources as they are?</summary>
  
  Yes. You can only have the `command.ts` file in the `/stubs` folder and customize it. The generator will use the default templates for the other resources.
</details>

<details>
  <summary>How can I use the default templates again!?</summary>
  
  Simply delete the `/stubs` folder or a specific resource file.
</details>


## Changes
- New CLI command `stub:publish`
- Tests
  - New code test coverage
  - Fixed old test code formatting
- New `template` method to get default or custom resource template 
- New files
  - `packages/cli/src/services/stub-publisher.ts`
  - `packages/cli/src/commands/stub/publish.ts`
- Removed `packages/cli/src/templates/index.ts` file

## Checks
- [x] Project Builds
- [x] Project passes tests and checks

## Additional information
- Closes #1074
